src/actions/waitFor/waitForAllRequestsComplete.ts

```diff
@@ -0,0 +1,105 @@
+import {LogEventType} from '../../constants/internal';
+import {getTestRunPromise} from '../../context/testRunPromise';
+import {getWaitForEventsState} from '../../context/waitForEventsState';
+import {E2edError} from '../../utils/error';
+import {setCustomInspectOnFunction} from '../../utils/fn';
+import {getFullPackConfig} from '../../utils/getFullPackConfig';
+import {log} from '../../utils/log';
+import {getPromiseWithResolveAndReject} from '../../utils/promise';
+import {RequestHookToWaitForEvents} from '../../utils/requestHooks';
+import {setReadonlyProperty} from '../../utils/setReadonlyProperty';
+import {getInitialIdsForAllRequestsCompletePredicate} from '../../utils/waitForEvents';
+
+import type {
+  AllRequestsCompletePredicateWithPromise,
+  RequestPredicate,
+  Void,
+} from '../../types/internal';
+
+/**
+ * Wait for the complete of all requests that satisfy the request predicate.
+ */
+export const waitForAllRequestsComplete = async (
+  predicate: RequestPredicate,
+  {firstRequestTimeout, timeout}: {firstRequestTimeout?: number; timeout?: number} = {},
+): Promise<void> => {
+  setCustomInspectOnFunction(predicate);
```

Comment:
I don't understand why it needed

Reply:
nodejs supports the ability to customize the result of the `util.inspect` function, which we use to print logs to the terminal and report: https://nodejs.org/dist/latest-v18.x/docs/api/util.html#utilinspectcustom.

Here we add some customization to additionally log the body of the function (for predicate functions, this usually makes sense, since they are small and do not contain closures).